dapp_stats/dapp_size_resolver.py

```diff
@@ -0,0 +1,144 @@
+import base64
+from dapp_runner.descriptor import DappDescriptor
+from dapp_runner.descriptor.dapp import PayloadDescriptor
+from dapp_runner.descriptor.parser import load_yamls
+import json
+from pathlib import Path
+from typing import Callable, Dict, List, Sequence, Tuple, TypedDict
+from urllib import request
+from urllib.error import HTTPError
+from yapapi.payload.vm import _DEFAULT_REPO_SRV, resolve_repo_srv
+
+
+class DappSizeResolverError(Exception):
+    pass
+
+
+class ResolvedPayloadSizes(TypedDict):
+    total_size: int
+    payloads: Dict[str, int]
+
+
+class DappSizeResolver:
+    @classmethod
+    def resolve_defined_payload_sizes(
+        cls, descriptor_paths: Sequence[Path]
+    ) -> Tuple[ResolvedPayloadSizes, List[str]]:
+        payloads_sizes = {}
+        errors = []
+        size_resolvers = {
+            "vm": cls._resolve_payload_size_for_vm_runtime,
```

Comment:
that's enough for now but if we wanted to support many different runtimes, I think it would have been better to implement resolvers as different classes with the same interface